schema/omega-services.async.api.yaml

```diff
@@ -1,525 +1,480 @@
-        transfer to PRO/TNA, giving those details of changes of ownership
-        and/or custody that may be significant in terms of authority,
-        integrity and interpretation.
-
-    relatedMaterial:
+    GeneralDatedDescription:
       type: object
-      description: |
-        A note identifying material that has a direct and significant
-        connection to the unit of description. This will usually be in the form
-        of references to material that dealt with similar functions at an
-        earlier, or later, period to the unit of description. References to
-        significant material dealing with similar subjects may also be
-        appropriate. The fact that related material references are always
-        reciprocal should help the judgment of how significant a reference is.
+      name: Dated Description
+      description: A dated description for any general Object
       properties:
-        linkHref:
-          type: string
-          description: The URL to this related material
-        linkText:
-          type: string
-          description: The text for the link for this related material
         description:
-          type: string
-          description: The description for this related material
+          $ref: '#/components/schemas/GeneralDescription'
+        date:
+          $ref: '#/components/schemas/ISODate'
 
-    separatedMaterial:
+    ISODate:
+      type: string
+      format: date
+      name: ISO date
+      description: A date string formatted according to ISO 8601
+
+    ErrorMessage:
       type: object
-      description: |
-        Identifies materials (and their location) that are organically part of
-        the unit of description that have been physically separated or removed.
+      name: Error Message
+      description: An error message consisting of a code and a description.
       properties:
-        linkHref:
-          type: string
-          description: The URL to this separated material
-        linkText:
-          type: string
-          description: The text for the link for this separated material
+        code:
+          $ref: '#/components/schemas/ErrorCode'
         description:
-          type: string
-          description: The description for this separated material
-
-    creatorIDs:
-      description: |
-        Identifies the organisation(s) or individual(s) responsible for
-        creating, accumulating or maintaining the records being described.
-        Incorporates the authority controlled corporate or personal name(s)
-        of the creator(s) of the records.
-      type: array
-      items:
-        type: string
-
-    assetVersion:
-      type: string
-      description: The version of an asset
+          $ref: '#/components/schemas/ErrorDescription'
 
-    assetOwnerId:
+    ErrorCode:
       type: string
-      description: Identifier of the owner of the asset
+      format: '[A-Z]{4}([1-9][0-9][0-9]|0[1-9][0-9]|00[1-9])'
```

Comment:
Is there a reason for including letters? Do we thinking there will be more than 999 different errors?

Reply:
It's semantic information for the developer. I thought we could use similar prefixes for similar classes of errors.

Reply:
OK. I'll see what I can come up with.

Reply:
Okay great. We need to create a `Developers Guide` document that will contain information for developers that wish to work with the Project Omega API Services. I will setup a document that we can both contribute to. We can then discuss some concrete error codes for the first services you are building.